### PR TITLE
crosscluster/logical: move job txn into helper

### PR DESCRIPTION
**Pull request by msbutler**
This pure refactor attempts to prevent any side effects from trickling into the
job creation txn.

Epic: none

Release note: none

### CODE REVIEW

**Comment by msbutler** [comment]
unrelated flake

File: pkg/crosscluster/logical/create_logical_replication_stmt.go [hunk 1]
```diff
@@ -281,63 +281,72 @@ func createLogicalReplicationStreamPlanHook(
...
+			if err != nil {
 				return err
 			}
-			if _, err := p.ExecCfg().JobRegistry.CreateAdoptableJobWithTxn(ctx, jr, jr.JobID, txn); err != nil {
+			dstTableDescs = append(dstTableDescs, dstTableDesc)
+		}
+
+		if buildutil.CrdbTestBuild {
+			if len(srcTableDescs) != len(dstTableDescs) {
+				panic("srcTableDescs and dstTableDescs should have the same length")
```
Comment (dt) [inline pkg/crosscluster/logical/create_logical_replication_stmt.go:333]: while you're here could you make this return an assertion error instead?

**Review by dt — APPROVED** [review]

File: pkg/crosscluster/logical/create_logical_replication_stmt.go (hunk 1 above)
Comment (msbutler) [reply pkg/crosscluster/logical/create_logical_replication_stmt.go]: Done.

**Pull request merged**